### ISSUE TITLE
bug: View automatically scrolls up on new data on iOS and Safari

### ISSUE THREAD

**Issue by 1x2x3x4x** [ionic-team/ionic-framework#22673]
# Bug Report

**Ionic version:**

[ ] **4.x**
[x] **5.x**

**Current behavior:**

Using Firestore and listening for Snapshot Changes, when new changes happen, if I scrolled down the page, it'll scroll back up instead of staying at the same position. It fails on iOS (both real device and simulator, latest iOS) and Safari browser. Works perfectly on Android and Chrome (iOS or Windows).

**Expected behavior:**

If I've scrolled down it should stick in the same position, even if new data comes, not scroll up like a hard refresh.

**Steps to reproduce:**

**Related code:**

```
See comment bellow.
```

**Other information:**

If I just do this everything works perfectly. So I guess it has to do with the cards or grid but I'm not really sure.
```
<section *ngIf="!showLoader" style="margin-bottom: 120px; margin-top: 250px;">
   <div *ngIf="deviceId" style="z-index: -100;">
      <div *ngFor="let telemetria of telemetrias">
        <h1>{{ telemetria.DATA1 }}</h1>
        <h1>{{ telemetria.DATA2 }}</h1>
	<h1>{{ telemetria.DATA3 }}</h1>
	<h1>{{ telemetria.DATA4 }}</h1>
	<h1>{{ telemetria.DATA5 }}</h1>
	<h1>{{ telemetria.DATA6 }}</h1>
	<h1>{{ telemetria.DATA7 }}</h1>
	<h1>{{ telemetria.DATA8 }}</h1>
	 <h1>{{ telemetria.DATA9 }}</h1>
	<h1>{{ telemetria.DATA10 }}</h1>
      </div>
   </div>
</section>
```

**Ionic info:**

```
Ionic:
   Ionic CLI                     : 6.12.2 (/usr/local/lib/node_modules/@ionic/cli)
   Ionic Framework               : @ionic/angular 5.5.2
   @angular-devkit/build-angular : 0.803.29
   @angular-devkit/schematics    : 10.0.8
   @angular/cli                  : 8.3.29
   @ionic/angular-toolkit        : 2.3.3

Cordova:
   Cordova CLI       : 10.0.0
   Cordova Platforms : ios 6.1.1
   Cordova Plugins   : cordova-plugin-ionic-keyboard 2.2.0, cordova-plugin-ionic-webview 4.2.1, (and 16 other plugins)
   
Utility:
   cordova-res                          : 0.15.2
   native-run (update available: 1.3.0) : 1.2.2

System:
   ios-sim : 8.0.2
   NodeJS  : v14.15.1 (/usr/local/bin/node)
   npm     : 6.14.9
   OS      : macOS Big Sur
   Xcode   : Xcode 12.2 Build version 12B45b
```

**Comment by 1x2x3x4x** [comment]
It's hard to provide a Stackblitz because I use **Firestore**, but it's easy to reproduce.

This only happens if you use **cards**. 


If you do this it'll _work_. The scroll will stay where it was before getting new data if you previously scrolled down.
```
<div *ngFor="let telemetria of telemetrias">
   <h1>x.date</h1>
   <h1>x.date</h1>
   <h1>x.date</h1>
   <h1>x.date</h1>
   <h1>x.date</h1>
   <h1>x.date</h1>
   <h1>x.date</h1>
   <h1>x.date</h1>
   <h1>x.date</h1>
</div>

```


If you do this it _won't work_ on iOS, it'll scroll automatically back up.
```
<div *ngFor="let telemetria of telemetrias">
   <ion-card>
         <h1>x.date</h1>
   </ion-card>
   <ion-card>
         <h1>x.date</h1>
   </ion-card>
   <ion-card>
         <h1>x.date</h1>
   </ion-card>
   <ion-card>
         <h1>x.date</h1>
   </ion-card>
   <ion-card>
         <h1>x.date</h1>
   </ion-card>
   <ion-card>
         <h1>x.date</h1>
   </ion-card>
</div>

```

I think it's a bug because it only happens on iOS/Safari. works great on Android/Chrome,

**Comment by 1x2x3x4x** [comment]
**Workaround:** use trackByFn.

https://angular.io/api/core/TrackByFunction
https://stackoverflow.com/questions/42108217/how-to-use-trackby-with-ngfor

**Comment by liamdebeasi** [comment]
Sorry for the delay. What version of iOS are you testing this on? I cannot reproduce this on the latest version of iOS 14.

**Comment by 1x2x3x4x** [comment]
Latest version of Big Sur with Safari and also iOS 14.3 on an iPhone 7. 

As I said, I used trackByFn to stop the scrolling back up. 
I can't test it again for now. 

However I thought you guys might be interested since it only happened on Safari and not Chrome. 

Thanks for the reply anyways.

**Comment by liamdebeasi** [comment]
I tested this on the iOS 14.4 beta, so it seems like if it was an issue it will be fixed in iOS 14.4. Glad to hear `trackByFn` resolves the issue on older versions of iOS. I am going to close this since I can't reproduce the issue. If the issue appears again feel free to open a new issue. Thanks!
